StorageErrorActivity: statvfs failed: EACCES (Permission denied)

Android 4.4
AntennaPod 1.5.2

```
java.lang.IllegalArgumentException: Invalid path: /storage/emulated/0/Android/data/de.danoeh.antennapod/files
	at android.os.StatFs.doStat(StatFs.java:46)
	at android.os.StatFs.<init>(StatFs.java:39)
	at android.support.design.R.getFreeSpaceAvailable(R.java:50285)
	at de.danoeh.antennapod.activity.StorageErrorActivity.access$lambda$0(StorageErrorActivity.java:2150)
	at de.danoeh.antennapod.activity.StorageErrorActivity$$Lambda$1.onClick(Unknown Source)
	at android.view.View.performClick(View.java:4466)
	at android.view.View$PerformClick.run(View.java:18542)
	at android.os.Handler.handleCallback(Handler.java:733)
	at android.os.Handler.dispatchMessage(Handler.java:95)
	at android.os.Looper.loop(Looper.java:136)
	at android.app.ActivityThread.main(ActivityThread.java:5097)
	at java.lang.reflect.Method.invokeNative(Native Method)
	at java.lang.reflect.Method.invoke(Method.java:515)
	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:785)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:601)
	at dalvik.system.NativeStart.main(Native Method)
Caused by: libcore.io.ErrnoException: statvfs failed: EACCES (Permission denied)
	at libcore.io.Posix.statvfs(Native Method)
	at libcore.io.ForwardingOs.statvfs(ForwardingOs.java:132)
	at android.os.StatFs.doStat(StatFs.java:44)
	... 15 more
```

--- a/app/src/main/java/de/danoeh/antennapod/activity/StorageErrorActivity.java
+++ b/app/src/main/java/de/danoeh/antennapod/activity/StorageErrorActivity.java
@@ -132,7 +132,8 @@ public class StorageErrorActivity extends AppCompatActivity {
         List<String> folders = new ArrayList<>(mediaDirs.length);
         List<CharSequence> choices = new ArrayList<>(mediaDirs.length);
         for(int i=0; i < mediaDirs.length; i++) {
-            if(mediaDirs[i] == null) {
+            File dir = mediaDirs[i];
+            if(dir == null || !dir.exists() || !dir.canRead() || !dir.canWrite()) {
                 continue;
             }
             String path = mediaDirs[i].getAbsolutePath();

--- a/app/src/main/java/de/danoeh/antennapod/preferences/PreferenceController.java
+++ b/app/src/main/java/de/danoeh/antennapod/preferences/PreferenceController.java
@@ -819,7 +819,8 @@ public class PreferenceController implements SharedPreferences.OnSharedPreferenc
         List<String> folders = new ArrayList<>(mediaDirs.length);
         List<CharSequence> choices = new ArrayList<>(mediaDirs.length);
         for(int i=0; i < mediaDirs.length; i++) {
-            if(mediaDirs[i] == null) {
+            File dir = mediaDirs[i];
+            if(dir == null || !dir.exists() || !dir.canRead() || !dir.canWrite()) {
                 continue;
             }
             String path = mediaDirs[i].getAbsolutePath();

